cpp/velox/operators/serializer/VeloxColumnarToRowConverter.cc

```diff
@@ -16,46 +16,64 @@
  */
 
 #include "VeloxColumnarToRowConverter.h"
+#include <velox/common/base/SuccinctPrinter.h>
+#include <cstdint>
 
 #include "memory/VeloxColumnarBatch.h"
+#include "utils/exception.h"
 #include "velox/row/UnsafeRowDeserializers.h"
 #include "velox/row/UnsafeRowFast.h"
 
 using namespace facebook;
 
 namespace gluten {
 
-void VeloxColumnarToRowConverter::refreshStates(facebook::velox::RowVectorPtr rowVector) {
-  numRows_ = rowVector->size();
+void VeloxColumnarToRowConverter::refreshStates(
+    facebook::velox::RowVectorPtr rowVector,
+    int64_t rowId,
+    int64_t memoryThreshold) {
+  auto vectorLength = rowVector->size();
   numCols_ = rowVector->childrenSize();
 
   fast_ = std::make_unique<velox::row::UnsafeRowFast>(rowVector);
 
   size_t totalMemorySize = 0;
   if (auto fixedRowSize = velox::row::UnsafeRowFast::fixedRowSize(velox::asRowType(rowVector->type()))) {
-    totalMemorySize += fixedRowSize.value() * numRows_;
+    GLUTEN_CHECK(
+        memoryThreshold > fixedRowSize.value(),
+        "spark.gluten.sql.columnarToRowMemoryThreshold(" + velox::succinctBytes(memoryThreshold) +
+            ") is too small, it can't hold even one row(" + velox::succinctBytes(fixedRowSize.value()) + ")");
```

Comment:
Let's not report exception here, just try to allocate the new size for one row. After all, user needs to increases the number.